server/services/reportEditService.ts

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable no-await-in-loop */
+import moment from 'moment'
+
+import type LocationService from './locationService'
+import AuthService from './authService'
+
+export default class ReportEditService {
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor(
+    private readonly locationService: LocationService,
+    private readonly authService: AuthService
+  ) {}
+
+  async constructChangesToView(username, reportSection, changes = []) {
+    const questionSet = this.#getQuestionSet(reportSection)
+    let response = []
+
+    if (reportSection?.section === 'incidentDetails') {
+      response = await this.#buildIncidentDetails(username, questionSet, changes)
+    }
+
+    return response
+  }
+
+  async persistChanges(data) {
+    if (data.reportSection.section === 'incidentDetails') {
+      // TODO
+      // - update the current report
+      // - create a new record in the report_edit table
+    }
+  }
+
+  validateReasonForChangeInput(input) {
+    if (!input.reason) {
+      return [
+        {
+          href: '#reason',
+          text: `Select the reason for changing ${input.reportSection.text}`,
+        },
+      ]
+    }
+    if (input.reason === 'anotherReasonForEdit' && !input.reasonText) {
+      return [
+        {
+          href: '#reasonText',
+          text: `Please specify the reason`,
+        },
+      ]
+    }
+    return []
+  }
+
+  private questionSets = {
```

Comment:
The questionSets object will eventually hold the questions for every section on the UOF report